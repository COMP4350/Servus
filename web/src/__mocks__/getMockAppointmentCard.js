function getMockAppointmentCard() {
    return {
        buyer: 'Risto',
        provider: 'Arveend',
        service_id: '1001',
        created_at: new Date(100),
        booked_time: new Date(500),
    }
}

export const mockAppointmentCard = getMockAppointmentCard();