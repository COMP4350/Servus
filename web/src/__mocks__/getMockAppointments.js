function getMockAppointments() {
    return [
        {
            buyer: 'Risto',
            provider: 'Arveend',
            service_id: '1001',
            created_at: new Date(100),
            booked_time: new Date(500),
        },
        {
            buyer: 'Arveend',
            provider: 'Risto',
            service_id: '1002',
            created_at: new Date(700),
            booked_time: new Date(1200),
        },
        {
            buyer: 'Caden',
            provider: 'Andy',
            service_id: '1003',
            created_at: new Date(1400),
            booked_time: new Date(1900),
        },
    ]
}

function getMockAppointmentsCookie() {
    return {
        username: 'testusername',
    }
}

export const mockAppointments = getMockAppointments();
export const mockAppointmentsCookie = getMockAppointmentsCookie();