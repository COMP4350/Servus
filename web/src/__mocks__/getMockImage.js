function getMockImage() {
    return {
        ownerUsername: 'riki',
        uploadDate: new Date(2012, 1, 1),
        imageName: 'example_image.jpeg',
        imageData: 'the/link/to/data',
        profilePicture: false,
    };
}

export const mockImage = getMockImage();
