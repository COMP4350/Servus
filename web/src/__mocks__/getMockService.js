function getMockService() {
    return {
        _id: '1001',
        name: "Risto's Painting",
        provider: 'Risto',
        description: 'Risto paints you',
        cost: 100,
        duration: '0100',
        availability: [{ weekday: 1, start_time: '0800', end_time: '1800' }],
        ratings: [{ username: 'arvind', rating: 5 }],
    };
}

export const mockService = getMockService();
