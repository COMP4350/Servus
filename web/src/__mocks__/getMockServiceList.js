function getMockServiceList() {
    return [
        {
            name: "Risto's Painting",
            provider: 'Risto',
            description: 'Risto paints you',
            cost: 100,
            duration: 60,
        },
        {
            name: "Reesto's Painting",
            provider: 'Reesto',
            description: 'Reesto paints yee',
            cost: 120,
            duration: 80,
        },
        {
            name: "Rasta's Painting",
            provider: 'Rasta',
            description: 'Rasta paints yaa',
            cost: 150,
            duration: 90,
        },
    ];
}

export const mockServiceList = getMockServiceList();
