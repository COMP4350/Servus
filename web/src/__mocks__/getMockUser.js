function getMockUser() {
    return {
        username: 'testuser',
        firstName: 'test',
        lastName: 'user',
        password: '123',
        bio: 'im a test user',
    };
}

export const mockUser = getMockUser();
