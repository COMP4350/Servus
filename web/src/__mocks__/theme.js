import { createMuiTheme } from '@material-ui/core/';

export default createMuiTheme({
    palette: {
        primary: {
            main: '#647AA3',
        },
        secondary: {
            main: '#EC5732',
        },
        default: {
            main: '#FFF2EB',
        },
    },
    background: {
        dark: '#151515',
        main: '#272727',
    },
    typographs: {
        fontFamily: ['Roboto'],
    },
});
