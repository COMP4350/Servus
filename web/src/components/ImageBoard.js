import React, { useState, useEffect } from 'react';
import { useParams } from 'react-router-dom';
import { useCookies } from 'react-cookie';
import axios from 'axios';
import {
    IconButton,
    FilledInput,
    GridList,
    GridListTile,
} from '@material-ui/core';
import PublishIcon from '@material-ui/icons/Publish';
import { makeStyles } from '@material-ui/core';

const useStyles = makeStyles(theme => ({
    root: {
        position: 'relative',
        width: '60%',
        height: '100%',
        overflow: 'hidden',
        backgroundColor: theme.background.dark,
    },
    inputRoot: {
        position: 'absolute',
        bottom: 0,
        right: 0,
        height: '75px',
        width: '75px',
    },
    btn: {
        position: 'absolute',
        bottom: 0,
        right: 0,
        height: '75px',
        width: '75px',
        'pointer-events': 'none',
        'z-index': 6,
    },
    image: {
        width: '33%',
        height: 'auto',
        'object-fit': 'cover',
    },
    gridList: {
        width: '100%',
        height: '100%',
    },
    icon: {
        width: '50px',
        height: '50px',
        color: 'white',
    },
}));

const inputStyles = makeStyles(() => ({
    root: {
        position: 'absolute',
        height: '75px',
        width: '75px',
        bottom: 0,
        right: 0,
        'z-index': '4',
    },
    input: {
        display: 'inline-block',
        'box-sizing': 'border-box',
        'border-radius': '20px',
        height: '75px',
        width: '75px',
        overflow: 'hidden',
        padding: '120px 0 0 0',
        'z-index': '4',
    },
}));

const ImageBoard = () => {
    const [images, setImages] = useState([]);
    const classes = useStyles();
    const inputclasses = inputStyles();
    const [cookies] = useCookies(['username']);
    const [change, setChange] = useState(false);
    let { targetUsername } = useParams();

    const getImages = async () => {
        const response = await axios.get(`/images/${targetUsername}`);
        let images = response.data.result;
        images.sort((a, b) => {
            a.uploadDate <= b.uploadDate ? 1 : -1;
        });
        setImages(images);
    };
    useEffect(() => {
        getImages();
    }, [targetUsername, change]);

    const getImageBoard = () => {
        console.log(images);
        return (
            <GridList cellHeight={200} className={classes.gridList} cols={3}>
                {images.map((oneImage, index) => {
                    return (
                        <GridListTile cols={1} key={index}>
                            <img
                                src={`${process.env.REACT_APP_API_HOST}/${oneImage.imageData}`}
                            />
                        </GridListTile>
                    );
                })}
            </GridList>
        );
    };

    const uploadImage = e => {
        let imageFormObj = new FormData();
        imageFormObj.append('imageName', 'multer-image-' + Date.now());
        imageFormObj.append('imageData', e.target.files[0]);
        imageFormObj.append('ownerUsername', targetUsername);

        // stores a readable instance of the image being uploaded using multer
        axios
            .post(`/images/upload`, imageFormObj)
            .then(() => {})
            .catch(err => {
                alert('Error while uploading image' + err);
            });
        setChange(!change);
    };
    return (
        <div className={classes.root}>
            {images ? getImageBoard() : null}

            {cookies.username && cookies.username === targetUsername ? (
                <div className={classes.inputRoot}>
                    <IconButton className={classes.btn} variant="contained">
                        <PublishIcon className={classes.icon} />
                    </IconButton>
                    <FilledInput
                        classes={inputclasses}
                        type="file"
                        onChange={e => uploadImage(e, 'multer')}></FilledInput>
                </div>
            ) : null}
        </div>
    );
};

export default ImageBoard;
