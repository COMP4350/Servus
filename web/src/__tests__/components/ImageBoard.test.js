import React from 'react';
import { act, render } from '@testing-library/react';
import ImageBoard from '../../components/ImageBoard';
import theme from '../../__mocks__/theme';
import { ThemeProvider } from '@material-ui/core/';
import { mockImage } from '../../__mocks__/getMockImage';

import mockAxios from 'axios';
jest.mock('react-router-dom', () => {
    return {
        __esModule: true,
        useParams: () => {
            return { targetUsername: 'yoyo' };
        },
    };
});

describe('ImageBoard', () => {
    it('renders correctly', async () => {
        const promise = Promise.resolve();
        mockAxios.get.mockImplementationOnce(() =>
            Promise.resolve({
                data: { result: [mockImage] },
            })
        );

        const { asFragment } = render(
            <ThemeProvider theme={theme}>
                <ImageBoard />
            </ThemeProvider>
        );
        expect(asFragment()).toMatchSnapshot();
        expect(mockAxios.get).toHaveBeenCalledWith('/images/yoyo');
        expect(mockAxios.get).toHaveBeenCalledTimes(1);

        await act(() => promise);
    });
});
