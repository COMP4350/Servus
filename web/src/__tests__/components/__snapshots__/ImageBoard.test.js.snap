// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageBoard renders correctly 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-root-1"
  >
    <div
      class="Toastify"
    />
    <ul
      class="MuiGridList-root makeStyles-gridList-5"
      style="margin: -2px;"
    />
  </div>
</DocumentFragment>
`;
